apps/portal/src/components/widgets/xcm/api/utils/toPreciseDecimals.ts

```diff
@@ -0,0 +1,9 @@
+import { AssetAmount } from '@galacticcouncil/xcm-core'
+import BigNumber from 'bignumber.js'
+
+export const toPreciseDecimals = (amount?: AssetAmount): string | undefined =>
+  amount
+    ? BigNumber(String(amount.amount ?? 0n))
+        .times(BigNumber(10).pow(-1 * (amount.decimals ?? 0)))
+        .toString()
+    : undefined
```

Comment:
seems like maybe `formatUnits` from viem can replace this

```suggestion
import { AssetAmount } from '@galacticcouncil/xcm-core'
import { formatUnits } from 'viem'

export const toPreciseDecimals = (amount?: AssetAmount): string | undefined =>
  amount ? formatUnits(amount.amount ?? 0n, amount.decimals ?? 1) : undefined
```